Update dependencies of the pubsub gem

VSpike opened the pull request:
The upstream fork has had some updates but our fork has diverged quite a lot. Update the Google Client gem.

We are seeing messages in the build of our FluentD setup like this:

```
Post-install message from google-gax:
*******************************************************************************
The google-gax gem is officially end-of-life and will not be updated further.

If your app uses the google-gax gem, it likely is using obsolete versions of
some Google Cloud client library (google-cloud-*) gem that depends on it. We
recommend updating any such libraries that depend on google-gax. Modern Google
Cloud client libraries will depend on the gapic-common gem instead.
*******************************************************************************
```

VSpike committed:
Update dependencies of the pubsub gem

The upstream fork has had some updates but our fork has diverged quite a
lot. Update the Google Client gem and Prometheus gem.

We are seeing messages in the build of our FluentD setup like this:

```
Post-install message from google-gax:
*******************************************************************************
The google-gax gem is officially end-of-life and will not be updated further.

If your app uses the google-gax gem, it likely is using obsolete versions of
some Google Cloud client library (google-cloud-*) gem that depends on it. We
recommend updating any such libraries that depend on google-gax. Modern Google
Cloud client libraries will depend on the gapic-common gem instead.
*******************************************************************************
```

asanord reviewed (approved):
That is a big bump. I am assuming this will be rolled out responsibly.

bencord0 commented:
I think you also need to update `.github/workflows/test.yaml` with the ruby version bump.

VSpike committed:
Update Github Action config, remove Travis config

Bump Ruby version in the Github Actions config.
Remove the Travis config as we are not using it.
Update the target version in Rubocop.

VSpike commented:
> I think you also need to update `.github/workflows/test.yaml` with the ruby version bump.

Thanks! Done.
